test/metabase/sync_database/analyze_test.clj

```diff
@@ -72,3 +79,78 @@
 (expect false (values-are-valid-emails? [100]))
 (expect false (values-are-valid-emails? ["true"]))
 (expect false (values-are-valid-emails? ["false"]))
+
+;; Tests to avoid analyzing hidden tables
+(defn count-unanalyzed-fields [db_id table-name]
+  (let [table-id (db/select-one-id Table, :db_id db_id, :active true :name table-name)]
+    (assert (pos? ;; don't let ourselves be fooled if the test passes because the table is
+                  ;; totally broken or has no fields. Make sure we actually test something
+             (db/count Field :table_id table-id)))
+    (db/count Field :last_analyzed nil :table_id table-id)))
+
+(defn get-latest-sync-time [db_id table-name]
+  (let [table-id (db/select-one-id Table, :db_id db_id, :active true :name table-name)]
+    (db/select-field :last_analyzed Field :table_id table-id)))
+
+
+(defn api-table-call [state table]
```

Comment:
I think all of our other functions that set some property of an object take the object as the first param. So might as well keep it consistent... instead you might consider doing something like

```clojure
(defn- set-table-visibility-type! [table visibility-type]
  ...)
```

(Calling the param `visibility-type` is clearer when doing things like looking at the argslist in the minibuffer because that's what we're setting, Tables don't have a state)